skimage/feature/blob.py

```diff
@@ -244,8 +254,19 @@ def blob_dog(image, min_sigma=1, max_sigma=50, sigma_ratio=1.6, threshold=2.0,
     """
     image = img_as_float(image)
 
+    # Gaussian filter requires that sequence-type sigmas have same
+    # dimensionality as image. This broadcasts scalar kernels
+    if isinstance(max_sigma, (int, float)):
+        max_sigma = np.full(len(image.shape), max_sigma, dtype=np.float)
+    if isinstance(min_sigma, (int, float)):
+        min_sigma = np.full(len(image.shape), min_sigma, dtype=np.float)
```

Comment:
Here, I suggest using `np.isscalar` instead of `isinstance`, because, for example, a NumPy int64 is not an instance of int (and one can get this type when, eg, grabbing a value from a NumPy array):

```python
In [7]: np.int_(4)
Out[7]: 4

In [8]: type(np.int_(4))
Out[8]: numpy.int64

In [9]: isinstance(np.int_(4), int)
Out[9]: False
```

(Weirdly, NumPy floats are instances of float.)

Additionally:

```suggestion
        min_sigma = np.full(image.ndim, min_sigma, dtype=float)
```

Little known factoid, `np.float` is just an alias for Python's `float`:

```
In [10]: np.float is float
Out[10]: True

In [11]: np.float_
Out[11]: numpy.float64
```